src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -487,6 +488,80 @@ def update_ag_redirect_configuration(cmd, instance, parent, item_name, redirect_
+    ApplicationGatewayRewriteRule, ApplicationGatewayRewriteRuleActionSet = cmd.get_models(
+        'ApplicationGatewayRewriteRule', 'ApplicationGatewayRewriteRuleActionSet')
+    if not request_headers and not response_headers:
+        raise CLIError('usage error: --response-headers HEADER=VALUE | --request-headers HEADER=VALUE')
+    ncf = network_client_factory(cmd.cli_ctx).application_gateways
+    ag = ncf.get(resource_group_name, application_gateway_name)
+    rule_set = next((x for x in ag.rewrite_rule_sets if x.name == parent_name), None)
+    if not rule_set:
+        raise CLIError("rule set '{}' not found.".format(parent_name))
+    new_rule = ApplicationGatewayRewriteRule(
+        name=item_name,
+        rule_sequence=sequence,
+        action_set=ApplicationGatewayRewriteRuleActionSet(
+            request_header_configurations=request_headers,
+            response_header_configurations=response_headers
+        )
+    )
+    _upsert(rule_set, 'rewrite_rules', new_rule, 'name')
+    _upsert(ag, 'rewrite_rule_sets', rule_set, 'name', warn=False)
+    return sdk_no_wait(no_wait, ncf.create_or_update, resource_group_name, application_gateway_name, ag)
+
+
+def update_ag_rewrite_rule(instance, parent, parent_name, item_name,
+                           sequence=None, request_headers=None, response_headers=None, cache_result=None):
+    return parent
+
+
+def create_ag_rewrite_rule_condition(cmd, resource_group_name, application_gateway_name, parent_name, item_name,
+                                     variable, no_wait=False, cache_result=False, pattern=None, ignore_case=None,
+                                     negate=None):
+    ApplicationGatewayRewriteRuleCondition = cmd.get_models(
+        'ApplicationGatewayRewriteRuleCondition')
+    ncf = network_client_factory(cmd.cli_ctx).application_gateways
+    ag = ncf.get(resource_group_name, application_gateway_name)
+    rule_set = next((x for x in ag.rewrite_rule_sets if x.name == parent_name), None)
+    if not rule_set:
+        raise CLIError("rule set '{}' not found.".format(parent_name))
+    rule = next((x for x in rule_set.rewrite_rules if x.name == item_name), None)
+    if not rule:
+        raise CLIError("rule '{}' in rule set '{}' not found.".format(item_name, parent_name))
+    if not rule.conditions:
+        rule.conditions = []
+    new_condition = ApplicationGatewayRewriteRuleCondition(
+        variable=variable,
+        pattern=pattern,
+        ignore_case=ignore_case,
+        negate=negate
+    )
+    _upsert(rule, 'condition', new_condition, 'variable')
+    _upsert(rule_set, 'rewrite_rules', rule, 'name', warn=False)
+    _upsert(ag, 'rewrite_rule_sets', rule_set, 'name', warn=False)
+    return sdk_no_wait(no_wait, ncf.create_or_update, resource_group_name, application_gateway_name, ag)
+
+
+def update_ag_rewrite_rule_condition(instance, parent, item_name, cache_result=None):
+    return parent
```

Comment:
I will delete this entire block. These commands are templated. These are the old custom commands I wrote.